### PR TITLE
Adding user agent to OAuth requests

### PR DESCRIPTION
**Pull request by khawkins**

### CODE REVIEW

File: libs/SalesforceOAuth/SalesforceOAuth/Classes/SFOAuthCoordinator.m [hunk 1]
```diff
@@ -516,29 +528,32 @@ - (void)beginUserAgentFlow {
...
     
     self.responseData = [NSMutableData dataWithLength:512];
-    NSString *url = [[NSString alloc] initWithFormat:@"%@://%@%@", 
-                                                     self.credentials.protocol, 
-                                                     self.credentials.domain, 
-                                                     kSFOAuthEndPointToken];
-	
-	NSMutableURLRequest *request = [[NSMutableURLRequest alloc] initWithURL:[NSURL URLWithString:url] 
-                                                                cachePolicy:NSURLRequestReloadIgnoringCacheData 
+    NSString *url = [[NSString alloc] initWithFormat:@"%@://%@%@",
```
Comment (khawkins) [inline libs/SalesforceOAuth/SalesforceOAuth/Classes/SFOAuthCoordinator.m]: This block is just whitespace changes.  Safely ignore.

File: libs/SalesforceOAuth/SalesforceOAuth/Classes/SFOAuthCoordinator.m [hunk 2]
```diff
@@ -516,29 +528,32 @@ - (void)beginUserAgentFlow {
...
+    if (self.userAgentForAuth != nil) {
+        [request setValue:self.userAgentForAuth forHTTPHeaderField:kHttpHeaderUserAgent];
+    }
     [request setHTTPShouldHandleCookies:NO];
     
     NSMutableString *params = [[NSMutableString alloc] initWithFormat:@"%@=%@&%@=%@&%@=%@",
-                                                                      kSFOAuthFormat, kSFOAuthFormatJson,
-                                                                      kSFOAuthRedirectUri, self.credentials.redirectUri,
-                                                                      kSFOAuthClientId, self.credentials.clientId];
+                               kSFOAuthFormat, kSFOAuthFormatJson,
```
Comment (khawkins) [inline libs/SalesforceOAuth/SalesforceOAuth/Classes/SFOAuthCoordinator.m]: Same here, just whitespace.

**Comment by bhariharan** [comment]
Looks good to me.

**Comment by wmathurin** [comment]
:+1:

**Pull request merged**